core/src/main/scala/org/apache/spark/deploy/LocalSparkCluster.scala

```diff
@@ -64,7 +64,8 @@ class LocalSparkCluster(
     /* Start the Workers */
     for (workerNum <- 1 to numWorkers) {
       val workerEnv = Worker.startRpcEnvAndEndpoint(localHostname, 0, 0, coresPerWorker,
-        memoryPerWorker, masters, null, Some(workerNum), _conf)
+        memoryPerWorker, masters, null, Some(workerNum), _conf,
+        conf.get(config.Worker.SPARK_WORKER_RESOURCE_FILE))
```

Comment:
Using the same resources file for different workers (whether local cluster or real cluster) really doesn't make sense if resources file is intent to have the cluster admin config with different resources. We can just pass None and only use discovery script for LocalSparkCluster since it is only used for test purpose. 

And for a real cluster, how about this: 

When user configs a resources file(even a discovery script configured concurrently), we just acquire resources from it and do not go through acquireResources() any more with the assuming that user has already configured different resources across workers. If not, then, we use discovery script and calls acquireResources() to make sure we get different  resources compare to others.

And we don't introduce a new configuration here, but just document more specific and rely on those file existence to decide which way we wanna go to. 

WDYT ?

Reply:
we could do that, but then again a cluster admin could write a discovery script to make sure different workers get different resources. Then they don't have to manually create the resourcesFile.    I also think there are some weird cases like you mention where you have both resources file and discovery script that wouldn't be obvious to the user what happens. one resource they separated but then another the discovery script didn't.  I realize these are corner cases but with a config it would be obvious exactly what is going to happen.

If we don't do the config then I think we should leave it as is and just document that the resourcesfile/discovery script for Workers/Driver in Standalone mode needs to have all the node resources or they need to configure a different resources Dir for each.

Reply:
> but then again a cluster admin could write a discovery script to make sure different workers get different resources. 

I don't know whether it is easy for admin to do this within a script. But I agree that it would be better to have a separate config option if we do expect a admin would do this.

So, let me add it later.

Reply:
I kind of forgot about this, we should probably document the fact you can't use separate resource file or discovery script so only way to do this  properly is with the coordination. I added an item to https://issues.apache.org/jira/browse/SPARK-27492 to make sure to document.